File: app/inventaire.ts
import {Personnage} from './personnage';
import {Item} from './item';


export class Inventaire {

  constructor(
    private _id: number,
    private _taille: number,
    private _items: Array<Item>,
    private _personnage: Personnage
               ) {}


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get taille(): number {
    return this._taille;
  }

  set taille(value: number) {
    this._taille = value;
  }

  get items(): Array<Item> {
    return this._items;
  }

  set items(value: Array<Item>) {
    this._items = value;
  }

  get personnage(): Personnage {
    return this._personnage;
  }

  set personnage(value: Personnage) {
    this._personnage = value;
  }
}

File: jdr-final/src/main/monapp/entity/Inventaire.java
package com.monapp.entity;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table
public class Inventaire {

	//CREATION DES VARIABLES :
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@JsonView(Views.Common.class)
	private int id;
	
	@Column
	@JsonView(Views.Common.class)
	private int taille;
	
	@OneToMany (mappedBy="inventaire", fetch=FetchType.EAGER)
	@JsonView(Views.Inventaire.class)
	private Set<Item> items = new HashSet<Item>();
	
	@OneToOne (mappedBy = "inventaire", fetch=FetchType.EAGER)
	@JsonView(Views.Inventaire.class)
	private Personnage personnage;
	
	// CONSTRUCTEURS :
	
	private Inventaire() {}
	
	private Inventaire(int taille) {
		this.taille = taille;
	}

	//GETTER ET SETTER :
	
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getTaille() {
		return taille;
	}

	public void setTaille(int taille) {
		this.taille = taille;
	}

	public Set<Item> getItems() {
		return items;
	}

	public void setItems(Set<Item> items) {
		this.items = items;
	}
	
	
}
